### PR TITLE
gh-112243: Don't include comments in f-string debug expressions

### PR DESCRIPTION
**Pull request by pablogsal**
* Issue: gh-112243

### CODE REVIEW

File: Parser/lexer/lexer.c [hunk 1]
```diff
@@ -112,13 +112,56 @@ set_fstring_expr(struct tok_state* tok, struct token *token, char c) {
...
+            return -1;
+        }
+
+        Py_ssize_t i = 0;
+        Py_ssize_t j = 0;
+
+        for (i = 0, j = 0; i < input_length; i++) {
+            if (tok_mode->last_expr_buffer[i] == '#') {
+                // Skip characters until newline or end of string
+                while (tok_mode->last_expr_buffer[i] != '\0') {
```
Comment (lysnikolaou) [inline Parser/lexer/lexer.c]: Is this a NULL-terminated string? I was expecting to loop until `last_expr_buffer == last_expr_end` here.

File: Parser/lexer/lexer.c [hunk 2]
```diff
@@ -112,13 +112,56 @@ set_fstring_expr(struct tok_state* tok, struct token *token, char c) {
...
+    PyObject *res = NULL;
 
-    PyObject *res = PyUnicode_DecodeUTF8(
-        tok_mode->last_expr_buffer,
-        tok_mode->last_expr_size - tok_mode->last_expr_end,
-        NULL
-    );
-    if (!res) {
+    // Check if there is a # character in the expression
+    int hash_detected;
```
Comment (lysnikolaou) [inline Parser/lexer/lexer.c]:
```suggestion
    int hash_detected = 0;
```

**Review by lysnikolaou — COMMENTED** [review]
Left a couple of comments. Also, the lint job fails.

**Comment by pablogsal** [comment]
@lysnikolaou fixed!

**Review by lysnikolaou — APPROVED** [review]
Yup, looks good!

**Pull request merged**